File: Java/342.java
class Solution {
    public boolean isPowerOfFour(int n) {
        return n > 0 && ((n & (n - 1)) == 0) && ((n & 0x55555555) != 0);
    }
}

class Solution2 {
    public boolean isPowerOfFour(int n) {
        while (n % 4 == 0) {
            n /= 4;
        }
        return n == 1;
    }
}

File: Python/342.py
import math


class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n == 1:
            return True

        if n <= 0:
            return False

        logarithm_base4 = math.log(n) / math.log(4)

        return logarithm_base4 == int(logarithm_base4)
